File: src/main/java/com/example/venka/javaspringsparkdemo/service/MessageServiceImpl.java
package com.example.venka.javaspringsparkdemo.service;

import com.example.venka.javaspringsparkdemo.dao.MessageDAO;
import com.example.venka.javaspringsparkdemo.model.Message;
import org.bson.types.ObjectId;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class MessageServiceImpl implements MessageService {

    private MessageDAO messageDAO;

    public MessageServiceImpl(@Autowired final MessageDAO messageDAO) {
        this.messageDAO = messageDAO;
    }

    @Override
    public void insert(Message message) {
        messageDAO.insert(message);
    }

    @Override
    public void delete(ObjectId id) {
        messageDAO.delete(id);
    }

    @Override
    public List<Message> getAll() {
        return messageDAO.getAll();
    }

    @Override
    public Message getById(ObjectId id) {
        return messageDAO.getById(id);
    }
}

File: src/main/kotlin/com/example/venka/kotlinspringsparkdemo/service/MessageServiceImpl.kt
package com.example.venka.kotlinspringsparkdemo.service

import com.example.venka.kotlinspringsparkdemo.dao.MessageDAO
import com.example.venka.kotlinspringsparkdemo.model.Message
import org.bson.types.ObjectId
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class MessageServiceImpl(@param:Autowired private val messageDAO: MessageDAO) : MessageService {

    override val all: List<Message>
        get() = messageDAO.all

    override fun insert(message: Message) {
        messageDAO.insert(message)
    }

    override fun delete(id: ObjectId) {
        messageDAO.delete(id)
    }

    override fun getById(id: ObjectId): Message {
        return messageDAO.getById(id)
    }
}
